airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -155,9 +156,32 @@ def request_params(self, stream_state: Mapping[str, Any], stream_slice: Mapping[
         params.update(**stream_slice)
         return params
 
+    def parse_response(self, response: requests.Response, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping]:
+        for record in super().parse_response(response, **kwargs):
+            if self.cursor_field in record:
+                record[self.cursor_field] = pendulum.from_timestamp(float(record[self.cursor_field]))
+
+            yield record
+
     def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
         current_stream_state = current_stream_state or {}
-        current_stream_state["ts"] = max(float(latest_record["ts"]), current_stream_state.get("ts", 0))
+        if isinstance(latest_record[self.cursor_field], DateTime):
+            latest_record[self.cursor_field] = latest_record[self.cursor_field].timestamp()
+
+        if isinstance(current_stream_state.get(self.cursor_field, 0), str):
+            try:
+                current_stream_state[self.cursor_field] = float(current_stream_state[self.cursor_field])
+            except ValueError:
+                current_stream_state[self.cursor_field] = pendulum.parse(current_stream_state[self.cursor_field]).timestamp()
+
+        elif isinstance(current_stream_state.get(self.cursor_field, 0), DateTime):
+            current_stream_state[self.cursor_field] = current_stream_state[self.cursor_field].timestamp()
+
+        current_stream_state[self.cursor_field] = max(
+            float(latest_record[self.cursor_field]), float(current_stream_state.get(self.cursor_field, 0))
+        )
+        current_stream_state[self.cursor_field] = pendulum.from_timestamp(current_stream_state[self.cursor_field])
+
         return current_stream_state
```

Comment:
we return state but the function annotation doesn't say it

Reply:
Done